### PR TITLE
feat(tap): Page Tap Support

### LINKED ISSUE #995: v0.161.0: Implement page/frame tap

### PR DESCRIPTION
**Pull request by avodovnik**
This PR adds support for [tap](https://playwright.dev/#version=v1.6.2&path=docs%2Fapi.md&q=pagetapselector-options).

closes #995

### CODE REVIEW

**Comment by kblok** [comment]
Can you pull from master? there were some changes in the Connection class.

**Review by kblok — CHANGES_REQUESTED** [review]

File: src/PlaywrightSharp.Tests/TapTests.cs [hunk 1]
```diff
@@ -0,0 +1,252 @@
...
+        public async Task ShouldNotSendMouseEventsTouchStartIsCanceled()
+        {
+            await Page.SetContentAsync(@"<div id=""a"" style=""background: lightblue; width: 50px; height: 50px"">a</div>");
+            await Page.EvaluateAsync(
+                @"() => {
+                    document.addEventListener('touchstart', t => t.preventDefault(), {passive: false});
+                }");
+
+            var handle = await TrackEvents("div");
+            await Page.TapAsync("div");
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp.Tests/TapTests.cs:71]: Is it?

https://github.com/microsoft/playwright/blob/master/test/tap.spec.ts#L50-L64

File: src/PlaywrightSharp.Tests/TapTests.cs [hunk 2]
```diff
@@ -0,0 +1,252 @@
...
+        public async Task ShouldNotSendMouseEventsWhenTouchEndIsCanceled()
+        {
+            await Page.SetContentAsync(@"<div id=""a"" style=""background: lightblue; width: 50px; height: 50px"">a</div>");
+            await Page.EvaluateAsync(
+                @"() => {
+                    document.addEventListener('touchend', t => t.preventDefault());
+                }");
+
+            var handle = await TrackEvents("div");
+            await Page.TapAsync("div");
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp.Tests/TapTests.cs:96]: See above comment. I can't seem to find a `body` reference in the original test.

File: src/PlaywrightSharp.Tests/TapTests.cs [hunk 3]
```diff
@@ -0,0 +1,252 @@
...
+            Assert.Null(touchEndResult);
+            Assert.Equal(40, touchStartResult.clientX);
+            Assert.Equal(60, touchStartResult.clientY);
+            Assert.Equal(1, touchStartResult.force);
+            Assert.Equal(0, touchStartResult.identifier);
+            Assert.Equal(40, touchStartResult.pageX);
+            Assert.Equal(60, touchStartResult.pageY);
+            Assert.Equal(1, touchStartResult.radiusX);
+            Assert.Equal(1, touchStartResult.radiusY);
+            Assert.Equal(0, touchStartResult.rotationAngle);
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp.Tests/TapTests.cs:210]: Yeah, good catch - though the value is `null` for us :)

File: src/PlaywrightSharp/Page.cs [hunk 4]
```diff
@@ -897,6 +901,10 @@ public Task<string> GetInnerTextAsync(string selector, int? timeout = null)
         public Task<string> GetTextContentAsync(string selector, int? timeout = null)
              => MainFrame.GetTextContentAsync(true, selector, timeout);
 
+        /// <inheritdoc />
+        public Task TapAsync(string selector, bool force = false, bool noWaitAfter = false, Modifier[] modifiers = null, Point? position = null, int timeout = 0)
+            => MainFrame.TapAsync(selector, force, noWaitAfter, modifiers, position, timeout);
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp/Page.cs:906]: Can you explain? Not sure I follow.

File: src/PlaywrightSharp.Tests/TapTests.cs [hunk 5]
```diff
@@ -0,0 +1,252 @@
...
+        public TapTests(ITestOutputHelper output) :
+                base(output)
+        {
+        }
+
+        /// <inheritdoc/>
+        public override async Task InitializeAsync()
+        {
+            await base.InitializeAsync();
+            Context = await Browser.NewContextAsync();
```
[the remark this answers is not in the document]
Comment (kblok) [reply src/PlaywrightSharp.Tests/TapTests.cs:27]: Where are we using this @avodovnik?

File: src/PlaywrightSharp.Tests/TapTests.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (kblok) [reply src/PlaywrightSharp.Tests/TapTests.cs:70]: ah ok. We can leave this as is.
Consider looking at the version tag we are implementing. I was looking at the spec file on the tag v1.6.2 https://github.com/microsoft/playwright/blob/v1.6.2/test/tap.spec.ts#L50

[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp.Tests/TapTests.cs]: Interesting. 

Yeah, I'd almost prefer we keep it as close to _current_ as we can, but maybe that's a conversation we should have.

File: src/PlaywrightSharp.Tests/TapTests.cs [hunk 6]
```diff
@@ -0,0 +1,273 @@
...
+        }
+
+        /// <playwright-file>tap.specs.ts</playwright-file>
+        /// <playwright-it>should work with modifiers</playwright-it>
+        [Fact(Timeout = PlaywrightSharp.Playwright.DefaultTimeout)]
+        public async Task ShouldWorkWithModifiers()
+        {
+            await Page.SetContentAsync("hello world");
+
+            var altKeyTask = Page.EvaluateAsync<JsonElement>(@"() =>
```
[the remark this answers is not in the document]
Comment (avodovnik) [reply src/PlaywrightSharp.Tests/TapTests.cs:165]: Great catch!

**Review by kblok — APPROVED** [review]

**Pull request merged**